Update protect-against-timing-attacks.md (#11617)
cloudflare/cloudflare-docs: content/workers/examples/protect-against-timing-attacks.md

To avoid timing attacks in your code, you can replace equality checks with the [`crypto.timingSafeEqual`](/workers/runtime-apis/web-crypto/#timingsafeequal) function in your Workers application.

To use this function, create a new [`TextEncoder`](/workers/runtime-apis/encoding/#textencoder) and encode the string values to instances of `ArrayBuffer` using [`encoder.encode`](/workers/runtime-apis/encoding/#methods). This is needed because `crypto.timingSafeEqual` compares `ArrayBuffer` instances, not strings. With the encoded values, replace the standard JavaScript equality check (`===`) with `crypto.timingSafeEqual`. Note that the strings must be the same length in order to compare to `timingSafeEqual`. The below code shows how to implement string equality checks with `crypto.timingSafeEqual`. Note that the example shown would apply to Typescript and Javascript:

```ts
const encoder = new TextEncoder();

> const a = new TextEncoder().encode("foo")
> const b = new TextEncoder().encode("bar")
> 
> if (string1.byteLength !== string2.byteLength) {

Reviewer: ```suggestion
const username = "foo";
const password = "bar";

if (username.length !== password.length) {
  // Minimise the possibility of a timing attack via how long encoding takes on the strings
}

const a = encoder.encode(username)
const b = encoder.encode(password)

if (a.byteLength !== b.byteLength) {
```

We can use the TextEncoder instance we created earlier for this, and amend the byteLength check to use the new variable names.

This adds a string length check, to minimise timing attacks from how long encoding takes, but we keep the byteLength check due to unicode.

```js
> "a".length
1
> "ä".length
1
> new TextEncoder().encode("a").length
1
> new TextEncoder().encode("ä").length
2
```

// The below code is vulnerable to timing attacks
// if (string1 === string2) { ... }

// You can replace it with `crypto.subtle.timingSafeEqual` by encoding the values
// you need to compare

> const a = encoder.encode("foo");
> const b = encoder.encode("bar");

Reviewer: ```suggestion
```

These can be removed as we already encoded the strings further up.

// You can replace it with `crypto.timingSafeEqual` by encoding the values
// you need to compare
const a = encoder.encode("foo");
const b = encoder.encode("bar");

> //let equal = crypto.timingSafeEqual(a, b)

Reviewer: ```suggestion
```

Author: I think that makes a lot of sense! 😊 I'll make the changes to the JS snippet.
